src/Orchard.Web/Modules/Orchard.DynamicForms/Drivers/QueryElementDriver.cs

```diff
@@ -174,5 +174,28 @@ public QueryElementDriver(IFormsBasedElementServices formsServices, IProjectionM
             var runtimeValue = element.RuntimeValue;
             return runtimeValue != null ? runtimeValue.Split(new[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries) : Enumerable.Empty<string>();
         }
+        
+        protected override void OnExporting(Query element, ExportElementContext context) {
+            base.OnExporting(element, context);
+            if (element.QueryId == null)
+                return;
+            var queryIdentityPart = _contentManager.Get<IdentityPart>(element.QueryId.Value);
```

Comment:
You should rely on _contentManager.GetItemMetadata().Identity instead - there are potentially other handlers that contribute to a content's identity.

Please check the import/export implementation of the `ProjectionElementDriver`, which solves the same issue that you;re solving here:

https://github.com/OrchardCMS/Orchard/blob/1.10.x/src/Orchard.Web/Modules/Orchard.Layouts/Drivers/ProjectionElementDriver.cs

Thanks!

Reply:
Thanks for the review. Updated the file.

Reply:
The following line:

`var queryIdentityPart = _contentManager.Get<IdentityPart>(element.QueryId.Value);`

Although it will work if `QueryPart`'s content type also includes the `IdentityPart`, the more robust way would be to not rely on that part to be present (since technically, it shouldn't rely on IdentityPart for its identity - any module could provide its identity). So how about do this instead:

`var queryPart = _contentManager.Get<QueryPart>(element.QueryId.Value);`

Reply:
I think it looks good otherwise.

Reply:
Yes, you are right. Applied it.